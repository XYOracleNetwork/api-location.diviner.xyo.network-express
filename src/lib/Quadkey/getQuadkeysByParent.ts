import { getParentQuadkey } from './getParentQuadkey'

export const getQuadkeysByParent = (heatmap: string[]): Record<string, string[]> => {
  const quadkeysByParent = heatmap.reduce<Record<string, string[]>>((acc, curr) => {
    const parent = getParentQuadkey(curr)
    acc[parent] = acc[parent] || []
    acc[parent].push(curr)
    return acc
  }, {})
  return quadkeysByParent
}
