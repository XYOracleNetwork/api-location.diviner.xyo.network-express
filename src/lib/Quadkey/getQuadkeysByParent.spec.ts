import { getQuadkeysByParent } from './getQuadkeysByParent'

const testData = [
  { input: ['00', '033', '0333'] },
  { input: ['10', '122', '1222'] },
  { input: ['20', '211', '2111'] },
  { input: ['30', '300', '3000'] },
  { input: ['40', '40', '40'] },
]

describe('getQuadkeysByParent', () => {
  it.each(testData)('converts the quadkey at the specified zoom level', (data) => {
    const actual = getQuadkeysByParent(data.input)
    expect(actual).toMatchSnapshot()
  })
})
