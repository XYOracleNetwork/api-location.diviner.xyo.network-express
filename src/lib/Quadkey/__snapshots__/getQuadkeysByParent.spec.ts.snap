// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 1`] = `
Object {
  "": Array [
    "0",
    "0",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 2`] = `
Object {
  "": Array [
    "1",
    "0",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 3`] = `
Object {
  "": Array [
    "2",
    "0",
  ],
}
`;

exports[`getQuadkeysByParent converts the quadkey at the specified zoom level 4`] = `
Object {
  "": Array [
    "3",
    "0",
  ],
}
`;
